packages/moonstone/Input/util.js

```diff
@@ -0,0 +1,59 @@
+import $L from '../internal/$L';
+
+/**
+ * Determines the `aria-label` for an Input
+ *
+ * @method
+ * @memberof moonstone/Input
+ * @param   {String}  prefix   Text to precede the value in the aria-label
+ * @param   {String}  type     `type` of the Input
+ * @param   {String}  [value]  Current value of the input
+ * @returns {String}           `aria-label` value
+ * @private
+ */
+const calcAriaLabel = function (prefix, type, value = '') {
+	const hint = $L('Input field');
+
+	if (type === 'password' && value) {
+		const character = value.length > 1 ? $L('characters') : $L('character');
+		value = `${value.length} ${character}`;
+	}
+
+	return `${prefix} ${value} ${hint}`;
+};
+
+/**
+ * Removes `<input>` related props from `props` and returns them in a new object.
+ *
+ * Useful when redirecting `<input>` related props from a non-input root element to the `<input>`
+ * element.
+ *
+ * @method
+ * @memberof moonstone/Input
+ * @param   {Object} props  Props object
+ * @returns {Object}        input related props
+ * @private
+ */
+const extractInputProps = function (props) {
+	const inputProps = {};
+	Object.keys(props).forEach(key => {
```

Comment:
Would it make more sense to put the supported props in an array and then `forEach` on them and if `props.hasOwnProperty()`, then delete?  Maybe it's more likely there are fewer keys on the props than there are suitable input attributes?